File: src/Kela.Domain/Entities/Workspace.cs
using Kela.Domain.Common;

namespace Kela.Domain.Entities;

public class Workspace : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public int? TeacherId { get; set; }
    public User? Teacher { get; set; }

    public ICollection<User> Students { get; set; } = new List<User>();
    public ICollection<Node> Nodes { get; set; } = new List<Node>();

}

File: laravel/app/Models/Workspace.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Workspace extends Model
{
    use SoftDeletes;

    protected $fillable = ['name', 'teacher_id'];

    public function teacher(): BelongsTo
    {
        return $this->belongsTo(User::class, 'teacher_id');
    }

    public function students(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'workspace_students', 'workspace_id', 'student_id');
    }

    public function nodes(): HasMany
    {
        return $this->hasMany(Node::class, 'workspace_id');
    }

    public function attendances(): HasMany
    {
        return $this->hasMany(Attendance::class, 'workspace_id');
    }
}
